packages/ui/src/components/vuestic-components/va-datepicker/components/VaDatePickerHeader/VaDatePickerHeader.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <div class="va-date-picker-header va-date-picker__header">
+    <va-icon name="chevron_left" size="small" @click="prev" />
+    <div class="header-text">
+      <slot name="header-text" v-bind="{ year, month, monthNames, view, changeView, switchView }">
+        <va-button flat @click="switchView" size="small">
+          <span class="mr-1">
+            <slot name="year" v-bind="{ year }">{{ year }}</slot>
+          </span>
+
+          <slot v-if="view !== 'year'" name="month" v-bind="{ month }">{{ monthNames[month] }}</slot>
+        </va-button>
+      </slot>
+    </div>
+    <va-icon name="chevron_right" size="small" @click="next" />
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, toRefs, PropType } from 'vue'
+import { useSyncProp } from '../../hooks/StatefulProp'
+import { VaDatePickerView } from '../../types/types'
+
+export default defineComponent({
+  name: 'VaDatePickerHeader',
+
+  emits: ['update:year', 'update:month', 'update:view'],
+
+  props: {
+    // Use default here to make year and month not undefined type
+    year: { type: Number, required: true, default: () => 0 },
+    month: { type: Number, required: true, default: () => 0 },
+    monthNames: { type: Array, required: true },
+    view: { type: String as PropType<VaDatePickerView>, required: true },
+  },
+
+  setup (props, { emit }) {
+    const { year, month, view } = toRefs(props)
+    const { syncProp: syncYear } = useSyncProp(year, 'year', emit, new Date().getFullYear())
+    const { syncProp: syncMonth } = useSyncProp(month, 'month', emit, new Date().getMonth())
+    const { syncProp: syncView } = useSyncProp(view, 'view', emit)
+
+    const next = () => {
+      if (view?.value === 'month') {
+        // If current month is December
+        if (syncMonth.value === 11) {
+          syncYear.value = syncYear.value + 1
+          syncMonth.value = 0
+        } else {
+          syncMonth.value = syncMonth.value + 1
+        }
+      } else if (view?.value === 'year') {
+        syncYear.value = syncYear.value + 1
+      }
+    }
+
+    const prev = () => {
```

Comment:
Hmm. Maybe date-fns? A lot of things could go wrong with vanilla date, especially if you factor in timezones.

Reply:
I don't use the Dates object here, because there will be a lot of issues with reactivity. I will make the code simpler to understand. But it will be much dirty working here with dates.